p2p/src/network.rs

```diff
@@ -66,7 +66,11 @@ impl<T: Pload, K: Kex + Sync, E: Enc + Sync> NetworkBaseHandle<T, K, E> {
     /// # Errors
     /// - If binding to address fail
     #[log(skip(public_key))]
-    pub async fn start(listen_addr: SocketAddr, public_key: PublicKey) -> Result<Self, Error> {
+    pub async fn start(
+        listen_addr: SocketAddr,
+        public_key: PublicKey,
+        private_key: PrivateKey,
```

Comment:
so why don't we use `KeyPair`?

Reply:
Public and private key lay separately in our configuration, so i decided to preserve it this way.

Reply:
And why pack them if we will just unpack them later.

Reply:
because in `KeyPair::new` maintains the invariant that these keys match, i.e. it makes sure they don't belong to 2 different pairs

Reply:
pairs should be transferred around in `KeyPair`